aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -159,14 +159,23 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
     // CUBLAS_STATUS_INVALID_VALUE error from cublasLtMatmulAlgoGetHeuristic.
     // self.dim() == 1 && result.dim() == 2 && self.sizes()[0] == mat2_sizes[1]
     // is to use lt interface only when self is bias.
+    // for cuda 11.4, cublasLtMatmul is activated
+    // the last two conditions is to skip 16b transA and non-trans-B having
+    // leading dim >> rows when they are sliced from a large tensor
+    // see fbcode/caffe2/test/test_linalg.py:test_corner_cases_of_cublasltmatmul
     useLtInterface = beta.toComplexDouble() == 1.0 && self.dim() == 1 &&
         result.dim() == 2 && self.sizes()[0] == mat2_sizes[1] &&
         self.is_contiguous() &&
         (scalar_type == at::ScalarType::Double ||
          scalar_type == at::ScalarType::Float ||
          scalar_type == at::ScalarType::Half ||
          scalar_type == at::ScalarType::BFloat16) &&
-        mat2_sizes[0] > 1 && mat2_sizes[1] > 1;
+        mat2_sizes[0] > 1 && mat2_sizes[1] > 1 &&
+        mat2_sizes[0] < 65535 && mat2_sizes[1] < 65535 &&
+        mat1_sizes[0] < 65535 && mat1_sizes[1] < 65535 &&
+        // avoid leaing dim >> rows bugs
+        ((mat1.strides()[0]==1 && mat1.strides()[1]==mat1_sizes[0]) || (mat1.strides()[1] == 1 && mat1.strides()[0] == mat1_sizes[1]) || (scalar_type != at::ScalarType::Half && scalar_type != at::ScalarType::BFloat16)) &&
+        ((mat2.strides()[0]==1 && mat2.strides()[1]==mat2_sizes[0]) || (mat2.strides()[1] == 1 && mat2.strides()[0] == mat2_sizes[1]) || (scalar_type != at::ScalarType::Half && scalar_type != at::ScalarType::BFloat16));
```

Comment:
Please, correct me if I am wrong, but do I understand it correctly, it seems like these checks correspond to
`(mat1 is col or row major) && (mat2 is col or row major) || (dtype is not in {Half, BFloat16})`, right? If so, how does it resolve the underlying issue described in the comment? I doubt Lt can be enabled when row/col-majorness is not satisfied, so the pass based on the dtype seems wrong to me.

Reply:
The condition seems to check dtype and whether the inputs are col or row major, it is not checking whether the lead dimension is larger than the rows.

Reply:
Would it be the second condition that checks that the other stride exactly matches the contig dimension? That lines up with the test taking slices of an existing tensor (that stride will no longer match sliced shape)

Reply:
leading dimension is later set from strides, so this check guarantees that the matrix is non-overlapping-and-dense